src/parent/reducers/reducers.js

```diff
@@ -17,18 +17,11 @@ function parentReducer(state = {}, action) {
         // All child actions are prefaced with 'CHILD'; if the incoming
         // action is for the child, let the child reducers handle the action
         if (subString === 'CHILD') {
-            const childState = state[action.windowName];
-            let newState = {};
-
-            if (childState) {
-                newState = Object.assign({}, state, {
-                    [action.windowName]: childReducers(childState, action)
-                });
-            } else {
-                newState = Object.assign({}, state, {
-                    [action.windowName]: childReducers({}, action)
-                });
-            }
+            const childState = state.childWindows && state.childWindows[action.windowName] || {};
+
+            const childWindows = Object.assign({}, state.childWindows);
+            childWindows[action.windowName] = childReducers(childState, action);
+            const newState = Object.assign({}, state, { childWindows });
```

Comment:
If we go with the combineReducers option above, then I think this could be simplified to something like this:

```js
return Object.assign({}, state, {
    [action.windowName]: childReducers(state[action.windowName], action)
});
```